minvws/nl-kat-coordination
# Support setting a custom JSON schema for copied boefjes

### Changes

This PR adds a "schema" field to the boefje database models so that we can also change the schema for copied (or new) boefjes. This was still lacking in the current APIs.

### Issue link

Closes https://github.com/minvws/nl-kat-coordination/issues/3052

### Demo

![image](https://github.com/user-attachments/assets/f03f6dbe-3e4d-40af-915a-ed15d12dd89f)

### QA notes

Again this is a migration mostly so: check out main, enable dns-records, check out this branch and verify that dns-records is still enabled:
```shell
git checkout main
# enable dns-records, change "test" to your own organisation id
curl -X PATCH -H 'Content-Type: application/json' --data-raw '{"enabled": true}' 'http://localhost:8003/v1/organisations/test/plugins/dns-records'
git checkout feature/add-json-schema-for-copied-boefjes
make kat
curl 'http://localhost:8003/v1/organisations/test/plugins/dns-records' | jq
# Expected: "enabled": true and a new "schema" field 
``` 

Edit: Verify that the migration ran successfully. The output of the command below should be `5be152459a7b`
```
docker compose exec -u postgres postgres /bin/bash -c 'psql -U $POSTGRES_USER -d katalogus -c "SELECT * FROM public.alembic_version;"'
```

---

### Code Checklist




---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
